contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import print_function
+#instructions to install googleapiclient can be found in this bot's readme file.
+import base64
+from googleapiclient import discovery
+from googleapiclient import errors
+import httplib2
+import logging
+from email.mime.text import MIMEText
+import mimetypes
+import os
+import re
+from oauth2client.file import Storage
+
+
+class GmailHandler(object):
+    '''
+    This plugin facilitates sending emails from a Gmail account
+    from within Zulip. It looks for messages starting with '@gmail'.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to send emails from a
+            personal gmail account. Users should preface
+            messages with "@gmail".
+
+            Enter messages in the following format:
+            "@gmail --recipient <recipient_email> --subject <subject> --label <label> --body
+            <body>". The --subject and --label tags are optional and can be left blank.
+
+            Before running this, make sure to create a bot and store its information
+            in ~/.zuliprc.
+            '''
+
+    def triage_message(self, message, client):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+        self.bot_email = client.email
+
+        # This next line of code is defensive
+        if message['display_recipient'] == 'gmail':
+            return False
+        return original_content.startswith('@gmail')
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        values = parse_input(original_content)
+        recipient = values[0]
+        subject = values[1]
+        label = values[2]
+        body = values[3]
+
+        service = get_service()
+        email = create_message(body, subject, recipient)
+        result = send_message(service, email)
+
+        label_result = -1
+        if label:
+            create_label(service, label)
+            label_id = get_label_id(service, label)
+            label_result = add_label(service, label_id, result['id'])
+
+        if ('SENT' in result['labelIds']):
```

Comment:
Please also remove parenthesis in if statements.
`if (something):` should be just `if something:`